client/client.go

```diff
@@ -574,6 +574,56 @@ func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
 	return allDelegations, nil
 }
 
+// RoleWithSignatures is a Role with its associated signatures
+type RoleWithSignatures struct {
+	Signatures []data.Signature
+	data.Role
+}
+
+// GetRepoRoleMetaInfo returns a list of RoleWithSignatures objects for this repo
+// This represents the latest metadata for each role in this repo
+func (r *NotaryRepository) GetRepoRoleMetaInfo() ([]RoleWithSignatures, error) {
```

Comment:
@cyli good point on the `Update` calls, I agree that the `GetRepoInfo` can't be two separate calls to this function and `ListTargets` (at least with how they're currently written).  I think what we can do here is call `ListTargets` to update and retrieve targets, and then follow the rest of `GetRepoRoleMetaInfo`'s logic without updating again: this can be our `GetRepoInfo` API and we can wrap `TargetsWithRoles` and `RoleWithSignatures` in a `RepoInfo` object.  How does that sound?

@endophage I don't think our API calls should rely on when other APIs were called, I think it's most expected to have each call update independently and always return the latest state at the time it was called.